key.ui/src/main/java/de/uka/ilkd/key/core/Watchdog.java

```diff
@@ -0,0 +1,116 @@
+package de.uka.ilkd.key.core;
+
+import java.awt.*;
+import java.util.Set;
+
+import de.uka.ilkd.key.util.ThreadUtilities;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Watchdog to monitor the state of the KeY system. If all worker and UI threads
+ * are waiting / blocked, a deadlock is reported.
```

Comment:
What is the purpose of this? Debugging? Is there any resolution when an error is detected?

Reply:
The purpose of this is to catch further bugs. Currently, the only resolution is some logging. It is not easy to notify the user since the UI thread will also be blocked.

Reply:
Ok, that's what I thought. Then it would be great if you could add this to the doc comment :)